### ISSUE TITLE
Amazfit Bip LineageOS 15.1 weather providers

### ISSUE THREAD

**Issue by github-k8n** [Freeyourgadget/Gadgetbridge#1075]
When using LineageOS 15.1 with a weather provider
e.g. https://www.reddit.com/r/LineageOS/comments/83yr54/lineageos_151_no_weather_provider/

Gadgetbridge does not provide any weather data to the connected Amazfit Bip.
The linked weather provider does work in cLock.

**Comment by ashimokawa** [comment]
Did  You setup a weather location? You have to do it in gadgetbridge settings. We cannot get the data from cLock - unfortunately

**Comment by ashimokawa** [comment]
Yes, best fit, you can connect your bip, then go to settings, enter Frankfurt, if you then go back and re-enter settings it might have changed. For me it changes to Hesse, whatever that is. 

;)

**Comment by github-k8n** [comment]
Did you try it on LineageOS 15.1 ? for me it doesn't change at all...

**Comment by ashimokawa** [comment]
@github-k8n 
I did not try on LineageOS 15.1 yet, only 14.1. Is there anything interesting in the logs?

**Comment by github-k8n** [comment]
nope, nothing interesting.. the only thing i get when entering a different location and going back to the settings overview is this:
04-19 17:07:05.559   511   667 D AutofillManagerService: onBackKeyPressed()
04-19 17:07:05.576  1705  1705 W InputEventReceiver: Attempted to finish an input event but the input event receiver has already been disposed.

**Comment by danielegobbetti** [comment]
Perhaps on Lineage 15 the only way to access these data is the new SDK?

**Comment by azuwis** [comment]
I got the some problem when using LineageOS 15.1.

Even cLock does not get weather data when using weather provider from https://www.reddit.com/r/LineageOS/comments/83yr54/lineageos_151_no_weather_provider/

As a workaround, I ported OmniJaws (another weather backend supported by GadgetBridge) to a standalone app, which should work on any modern Android ROM, custom or not.

Prebuild apk https://github.com/azuwis/android_OmniJaws/releases, or build your own using Gradle/Android studio.

Note, in my test, I had to whitelist OmniJaws in Battery Optimization to make the weather update work.

**Comment by ashimokawa** [comment]
@azuwis
Wow, this is great. Would be cool to have it in fdroid.

To make use of the lineage weather provider in 15.1 it seems we need to support both the old cm SDK (for lineage 13/14.1) and the new lineage SDK (for lineage 15.1)

**Comment by azuwis** [comment]
Is it OK to publish a system app from custom ROM to fdroid? I've no experience in fdroid publishing.

**Comment by ashimokawa** [comment]
@azuwis
If you do not violate/change the license of the original authors (which I assume didn't happen here) it should be OK.

There are also other forked system apps in f-droid.

I will have a look at your fork later.

**Comment by IzzySoft** [comment]
If (or until) you can make it to the main repo: if you'd sign the `.apk`, I could serve it via [my repo](https://apt.izzysoft.de/fdroid/repo). Of course you should make sure about the license thing Andreas just mentioned :wink:

**Comment by danielegobbetti** [comment]
@azuwis AFAIK in order to get your app included you have to create an issue [in this f-droid repo](https://gitlab.com/fdroid/rfp/issues). The policies are [here](https://f-droid.org/docs/Inclusion_Policy/?title=Inclusion_Policy) but you will find more detailed information in the issue template.

**Comment by azuwis** [comment]
From f-droid include policy https://f-droid.org/en/docs/Inclusion_Policy/

> The software should use its own unique Android package ID. Where the application is a fork of another (even one not included in the F-Droid repository) it must have a new ID, different from the original.

After read https://developer.android.com/studio/build/application-id, it's still not clear to me:

* Should I only change application ID, leave package name unchanged, or change both?
* Should I change application ID completely (e.g. `nodomain.omnijaws`), or appending suffix (e.g. `org.omnirom.omnijaws.standalone`)?
* Does changing package ID or package name effect functionality? Like the content provider uri `content://org.omnirom.omnijaws.provider/weather`.

**Comment by ashimokawa** [comment]
@azuwis 
You should override only the application id in build.gradle, I would not use org.omnirom, because it then looks like it is from them. Our package is named nodomain because we did not have freeyourgadget.org back then.

The content provider url is untouched by this change, it should stay, so that our (and others) code just works. We also do this by using pebbles content uri.

**Comment by azuwis** [comment]
I have a suggestion, how about we change the application ID to `org.freeyourgadget.omnijaws`, and maintain it under https://github.com/Freeyourgadget? It should not make too much effect to maintain.

**Comment by danielegobbetti** [comment]
@azuwis personally I agree to your proposal

**Comment by ashimokawa** [comment]
@azuwis 
I also agree

**Comment by azuwis** [comment]
My OmniJaws fork (https://github.com/azuwis/OmniJaws) is almost ready for Freeyourgadget, but encounter a problem.

After changing the applicationID (https://github.com/azuwis/OmniJaws/commit/d681be3d4a5e53121bb3933406a8b808de75b6e6), `Last update time` in setting activity always gives `Disabled`, weather data still can be fetched from content provider by `adb shell content query --uri content://org.omnirom.omnijaws.provider/weather`, but no longer updates.

Another problem (whether changing applicationID or not) is when targeting SDK 26+, widget does not update itself. May related to https://developer.android.com/about/versions/oreo/background Logcat:

    BroadcastQueue: Background execution not allowed: receiving Intent { act=org.omnirom.omnijaws.WEATHER_UPDATE flg=0x10 } to org.freeyourgadget.omnijaws/org.omnirom.omnijaws.widget.WeatherAppWidgetProvider

**Comment by danielegobbetti** [comment]
@azuwis since fdroid can also build from any branch, and I see that the upstream repo is using branches for android versions, could you try to reach out to them and see if they are interested in merging your changes in a new ``standalone-XX`` branch? 
That way fdroid could build from the upstream repo and all issues would be gone.

**Comment by ashimokawa** [comment]
@azuwis 
@danielegobbetti 
The question is how other apps detect the provider, do they also ask the package manager? If there is no other way, we could also ask for org.freeyourgadget.omnijaws, but that would limit the provider to Gadgetbridge..

**Comment by ashimokawa** [comment]
@azuwis 
I have created a clean fork of OmniJaws  Freeyourgadget/ - feel free to push your changes there.

**Comment by ashimokawa** [comment]
@github-k8n 
FYI, I tried to support the Lineage Weather Provider via their SDK, but since they have this "copy out binary .jar in your folder"-mentality I think we might be stuck here...

**Comment by azuwis** [comment]
I've force pushed to `standalone` and `wip` branches with changes that keep upstream project file structure.

This should make reviewing commits and merging future upstream commits much easier.
